File: 리트코드/0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        PriorityQueue<Point> pq = new PriorityQueue<>(
            (p1,p2) -> {
                int p1Distance = p1.x* p1.x + p1.y * p1.y;
                int p2Distance = p2.x* p2.x + p2.y * p2.y;
                return p1Distance - p2Distance;
            }
        );

        for(int[] point : points){
            pq.add(new Point(point[0],point[1]));
        }

        int[][] result = new int[k][2];
        for(int i = 0; i < k; i++){
            Point point = pq.poll();
            result[i][0] = point.x;
            result[i][1] = point.y;
        }

        return result;
    }

    class Point{
        int x;
        int y;

        public Point(int x, int y){
            this.x = x;
            this.y = y;
        }
    }
}

File: 리트코드/0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.kt
class Solution {
    fun kClosest(points: Array<IntArray>, k: Int): Array<IntArray> {
        val pq = PriorityQueue<Point>{p1,p2 ->
            val distance1 = p1.x *p1.x + p1.y * p1.y
            val distance2 = p2.x *p2.x + p2.y * p2.y
            distance1 - distance2
        }
        for (point in points){
            pq.add(Point(point[0],point[1]))
        }
        val results: Array<IntArray> = Array(k) { IntArray(2){0} }

        for (i in 0 until k) {
            val point = pq.poll()
            results[i][0] = point.x
            results[i][1] = point.y
        }
        return results
    }
    class Point(var x:Int, var y: Int){

    }
}